lib/tests/maintainers.nix

```diff
@@ -20,16 +20,20 @@ let
         ];
       }).config;
 
-      checkGithubId = lib.optional (checkedAttrs.github != null && checkedAttrs.githubId == null) ''
+      checks = lib.optional (checkedAttrs.github != null && checkedAttrs.githubId == null) ''
         echo ${lib.escapeShellArg (lib.showOption prefix)}': If `github` is specified, `githubId` must be too.'
         # Calling this too often would hit non-authenticated API limits, but this
         # shouldn't happen since such errors will get fixed rather quickly
         info=$(curl -sS https://api.github.com/users/${checkedAttrs.github})
         id=$(jq -r '.id' <<< "$info")
         echo "The GitHub ID for GitHub user ${checkedAttrs.github} is $id:"
         echo -e "    githubId = $id;\n"
+      '' ++ lib.optional (checkedAttrs.email == null && checkedAttrs.github == null && checkedAttrs.matrix == null) ''
+        echo ${lib.escapeShellArg (lib.showOption prefix)}': At least one of `email`, `github` or `matrix` must be specified, so that users know how to reach you.'
+      '' ++ lib.optional (checkedAttrs.email != null && lib.hasSuffix "noreply.github.com" checkedAttrs.email) ''
```

Comment:
How do these `noreply` emails work? Are they just a blackhole or do they get forwarded to the GitHub users actual email?

Reply:
They're a blackhole.

Reply:
I see. In any case I think this check here is way too specific to have the desired effect. If we want to avoid people using unreachable emails, then we should do email validation (I'd love that). If we don't want to do that then it's better imo to just live with the fact that not all emails can be reached.

Reply:
> email validation

No idea what that would entail.

Anyway, this is kind of a false dilemma: `noreply.github.com` is an obvious "statically known" unreachable domain that many people use because they think they have to supply one, there's no reason not to do this even if we don't go all the way to email validation(?).

Reply:
Fair enough, that makes sense!